File: frontend/src/types/draft.ts
export interface SegmentVisual {
  type: 'flow' | 'command'
  items: string[]
}

export interface VideoSegment {
  number: number
  headline: string
  summary: string
  impact: string
  action: string
  slide_title: string
  narration: string
  title_ja?: string
  category?: string
  visual?: SegmentVisual | null
  rank_reason?: string
}

export interface VideoPlanDraft {
  title: string
  week_label: string
  thumbnail_text: string
  thumbnail_text_candidates?: string[]
  intro: string
  segments: VideoSegment[]
  outro: string
  slide_outline: string[]
  narration_script: string
  description: string
  hashtags: string[]
  reference_urls: string[]
  total_items: number
  generated_at: string
  hook?: string
  title_candidates?: string[]
}

export interface UsedNewsEntry {
  id: string
  used_at: string
}

export interface VideoArtifact {
  id: string
  title: string
  created_at: string
  draft_generated_at: string
  total_items: number
  duration_seconds: number
  video_path: string
  subtitles_path: string
  slide_count: number
  chapters?: string
  youtube_description?: string
  thumbnail_path?: string
  title_candidates?: string[]
  thumbnail_text_candidates?: string[]
}

export interface VideoArtifactList {
  items: VideoArtifact[]
}

File: backend/app/schemas/draft.py
from typing import Literal

from pydantic import BaseModel


class SegmentVisual(BaseModel):
    # flow: 攻撃フロー等の3ステップ図 / command: コマンド・利用イメージのコードブロック
    type: Literal["flow", "command"]
    items: list[str]


class VideoSegment(BaseModel):
    number: int
    headline: str
    summary: str
    impact: str
    action: str
    slide_title: str
    narration: str
    source: str = ""
    title_ja: str = ""
    category: str = ""
    visual: SegmentVisual | None = None


class VideoPlanDraft(BaseModel):
    title: str
    title_candidates: list[str] = []
    week_label: str
    thumbnail_text: str
    thumbnail_text_candidates: list[str] = []
    hook: str = ""
    intro: str
    segments: list[VideoSegment]
    outro: str
    slide_outline: list[str]
    narration_script: str
    description: str
    hashtags: list[str]
    reference_urls: list[str]
    total_items: int
    generated_at: str
